sim/midas/src/main/cc/bridges/termination.cc

```diff
@@ -0,0 +1,65 @@
+#ifdef TERMINATIONBRIDGEMODULE_struct_guard
+
+#include "termination.h"
+#include <iostream>
+
+termination_t::termination_t(
+  simif_t* sim, 
+  std::vector<std::string> &args,
+  TERMINATIONBRIDGEMODULE_struct * mmio_addrs,
+  unsigned int num_messages,
+  unsigned int* is_err,
+  const char* const* msgs): 
+  bridge_driver_t(sim), 
+  mmio_addrs(mmio_addrs), 
+  num_messages(num_messages), 
+  is_err(is_err),
+  msgs(msgs)
+{
+  //tick-rate to decide sampling rate of MMIOs per number of ticks
+  std::string tick_rate_arg = std::string("+tick-rate=");
+  for (auto &arg: args) {
+    if (arg.find(tick_rate_arg) == 0) {
+      char *str = const_cast<char*>(arg.c_str()) + tick_rate_arg.length();
+      uint64_t tick_period = atol(str);
+      this->tick_rate = tick_period;
+    }
+  }
+}
+
+termination_t::~termination_t() {
+  free(this->mmio_addrs);
+}
+
+void termination_t::tick() {  //reads the MMIOs at tick-rate
+  if(tick_counter == tick_rate) {
+    if (read(this->mmio_addrs->out_status)) {
+      int msg_id = read(this->mmio_addrs->out_terminationCode);
+      this->fail = this->is_err[msg_id];
+      test_done = true;
+      if(msg_id < this->num_messages) {
```

Comment:
Thanks for fixing the indentation. It's easier to review this. 

I think you should either make this comparison an assert, or change line 44 to say msg_id / error code was out of bounds, and call exit().

Reply:
And you need to move the assert / bounds check above the index on line 38.